src/components/Search/server-actions.tsx

```diff
@@ -52,6 +51,64 @@ export interface AskAnswerResult {
     hasAnswer: boolean;
 }
 
+export async function searchSiteContent(args: {
+    query: string;
+    siteSpaceIds?: string[];
+    cacheKey?: string;
+}): Promise<OrderedComputedResult[]> {
+    const { siteSpaceIds = [], query, cacheKey } = args;
+    const pointer = getContentPointer();
+
+    if ('siteId' in pointer && 'organizationId' in pointer) {
+        const searchResults = await api.searchSiteContent(
+            pointer.organizationId,
+            pointer.siteId,
+            query,
+            siteSpaceIds,
```

Comment:
Why do we need to pass the space IDs since the API backend should know them?

Reply:
Understanding the rest, I think we should have it typed as `siteSpaceIds: string[] | undefined`; the case `[]` to indicate all is weird

Reply:
This isn't all siteSpaceIds.. it's just the spaceIds that we want to search in.

Reply:
Basically this adds a new feature that is not being used.. we can select specific spaces using this endpoint rather than one or all.